src/ERC4973.sol

```diff
@@ -5,88 +5,105 @@ import {SignatureChecker} from
 
 bytes32 constant AGREEMENT_HASH =
   keccak256("Agreement(address active,address passive,bytes metadata)");
 
 /// @notice Reference implementation of EIP-4973 tokens.
 /// @author Tim Daubenschütz, Rahul Rumalla (https://github.com/rugpullindex/ERC4973/blob/master/src/ERC4973.sol)
-abstract contract ERC4973 is EIP712, ERC165, IERC721Metadata, IERC4973 {
+abstract contract ERC4973 is
+  EIP712,
+  ERC721,
+  ERC721URIStorage,
+  IERC4973,
+  IERC5192
+{
   using BitMaps for BitMaps.BitMap;
 
   BitMaps.BitMap private _usedHashes;
 
-  string private _name;
-  string private _symbol;
-
-  mapping(uint256 => address) private _owners;
-  mapping(uint256 => string) private _tokenURIs;
-  mapping(address => uint256) private _balances;
+  modifier notSupported() {
+    revert("notSupported: Soulbound NFT EIP-5192");
+    _;
+  }
 
-  constructor(string memory name_, string memory symbol_, string memory version)
-    EIP712(name_, version)
-  {
-    _name = name_;
-    _symbol = symbol_;
+  function getApproved(uint256) public pure override returns (address) {
+    return address(0x0);
   }
 
-  function supportsInterface(bytes4 interfaceId)
+  function isApprovedForAll(address, address)
     public
-    view
-    virtual
+    pure
     override
     returns (bool)
   {
-    return interfaceId == type(IERC721Metadata).interfaceId
-      || interfaceId == type(IERC4973).interfaceId
-      || super.supportsInterface(interfaceId);
+    return false;
   }
 
-  function name() public view virtual override returns (string memory) {
-    return _name;
-  }
+  function approve(address, uint256) public override notSupported {}
+
+  function setApprovalForAll(address, bool) public override notSupported {}
+
+  function transferFrom(address, address, uint256) public override notSupported {}
+
+  function safeTransferFrom(address, address, uint256)
+    public
+    override
+    notSupported
+  {}
+
+  function safeTransferFrom(address, address, uint256, bytes memory)
+    public
+    override
+    notSupported
+  {}
+
+  constructor(string memory name, string memory symbol, string memory version)
+    EIP712(name, version)
+    ERC721(name, symbol)
+  {}
 
-  function symbol() public view virtual override returns (string memory) {
-    return _symbol;
+  function _burn(uint256 tokenId) internal override (ERC721, ERC721URIStorage) {
+    super._burn(tokenId);
   }
 
   function tokenURI(uint256 tokenId)
     public
     view
-    virtual
-    override
+    override (ERC721, ERC721URIStorage)
     returns (string memory)
   {
-    require(_exists(tokenId), "tokenURI: token doesn't exist");
-    return _tokenURIs[tokenId];
+    return super.tokenURI(tokenId);
   }
 
-  function unequip(uint256 tokenId) public virtual override {
-    require(msg.sender == ownerOf(tokenId), "unequip: sender must be owner");
-    _usedHashes.unset(tokenId);
-    _burn(tokenId);
+  function locked(uint256 tokenId) external view returns (bool) {
```

Comment:
Memo to myself: It would probably be better if there were a separation between `function locked() external` and `function _locked() internal` as with OZ methods, where an integrating user can then overrrite it.